staging/src/k8s.io/apimachinery/pkg/util/sets/byte.go

```diff
@@ -14,86 +14,50 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Code generated by set-gen. DO NOT EDIT.
-
 package sets
 
-import (
-	"reflect"
-	"sort"
-)
-
-// sets.Byte is a set of bytes, implemented via map[byte]struct{} for minimal memory consumption.
+// Byte is a set of bytes, implemented via map[byte]struct{} for minimal memory consumption.
 type Byte map[byte]Empty
 
 // NewByte creates a Byte from a list of values.
 func NewByte(items ...byte) Byte {
-	ss := make(Byte, len(items))
```

Comment:
If you change these generated sets, then we won't be able to run a benchmark? I suspect the generic code may actually be slower, we should confirm how much.

The `cast` trick makes it easy to turn one of these types into a generic set if needed, without modifying the generated implementation anyway, right?

Reply:
Yes, this will not able to run the benchmark of both. I should just expose the `cast` trick and keep generated implementation the old way. 

User can cast generated set to generic set on demand.

Reply:
I would add benchmarks in the initial commit of this PR, run them (`go test -count=10 ... >/tmp/before.log`), then change the implementation and run them again, then compare (`benchstat /tmp/before.log /tmp/after.log`).

> User can cast generated set to generic set on demand.

Why would they do that? IMHO one of the goals for this PR should be to simplify the implementation by getting rid of the code generator and instead use Go generics.

Reply:
> I would add benchmarks in the initial commit of this PR, run them (go test -count=10 ... >/tmp/before.log), then change the implementation and run them again, then compare (benchstat /tmp/before.log /tmp/after.log).

I'm not sure if I understand you correctly. Are you referring to me doing benchmarks of different implementations in each of the two commits and then comparing the results?

But we ultimately need somewhere to record the results of the comparison. Maybe we can record the result directly in this pr conversation?

> IMHO one of the goals for this PR should be to simplify the implementation by getting rid of the code generator and instead use Go generics.

Goals the same.